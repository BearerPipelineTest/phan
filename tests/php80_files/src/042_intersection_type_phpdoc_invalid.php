<?php

/**
 * Phan should warn and not crash when passed invalid combinations of types in intersection types.
 * (the error messages in this test may change in the future)
 * @param Closure(int&string):(stdClass&ArrayObject) $value
 * @param int&null $other
 */
function test42($value, $other, object $x) {
    $value($x); // should not warn
    $value(1); // should warn
    $value(new stdClass()); // should warn
    $value($other); // should warn
    '@phan-debug-var $other';
}
test42(
    function (int $x): stdClass {
        return (object)['x' => $x];
    },
    null,
    new stdClass()
);
