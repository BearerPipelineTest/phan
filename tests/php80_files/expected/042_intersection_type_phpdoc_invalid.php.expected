%s:10 PhanTypeMismatchArgumentProbablyReal Argument 1 ($p0) is $x of type object but Closure(int|string):(\ArrayObject|\stdClass) takes int|string (no real type) defined at %s:9 (the inferred real argument type has nothing in common with the parameter's phpdoc type)
%s:12 PhanTypeMismatchArgumentProbablyReal Argument 1 ($p0) is new stdClass() of type \stdClass but Closure(int|string):(\ArrayObject|\stdClass) takes int|string (no real type) defined at %s:9 (the inferred real argument type has nothing in common with the parameter's phpdoc type)
%s:13 PhanDebugAnnotation @phan-debug-var requested for variable $other - it has union type int|null
